tests/tracer/test_processors.py

```diff
@@ -216,3 +218,85 @@ def test_aggregator_partial_flush_2_spans():
     assert writer.pop() == [child1, child2]
     parent.finish()
     assert writer.pop() == [parent]
+
+
+def test_trace_top_level_span_processor_partial_flushing():
```

Comment:
These tests using a `Tracer` are probably better suited to be in `test_tracer.py` but I don't have a strong opinion